app/js/components/homePage/manageApps/manageApps.jsx

```diff
@@ -54,22 +58,80 @@ export default class ManageApps extends React.Component {
     const url = location.href.split('/')[3];
     let apiBaseUrl = `/${applicationDistribution}/${url}/ws/rest`;
     this.requestUrl ='/owa/addapp';
+    let that = this;
 
     const addonFile = new FormData();
     addonFile.append('file', document.getElementById('fileInput').files[0]);
-    $.ajax({
+  
+    const response = $.ajax({
+      xhr: function(){
+        let xhrRequest = $.ajaxSetup().xhr();
+        if(xhrRequest.upload){
+          xhrRequest.upload.addEventListener('progress',that.handleProgress, false);
+        }
+        return xhrRequest;
+       
+      },
       type: "POST",
       url: `https:/${apiBaseUrl}${this.requestUrl}`,
       data: addonFile,
       contentType: false,
       processData: false,
       cache: false,
       success: function (result) {
+        this.setState({
+          msgBody: "App has been successfully installed",
+          msgType: "success",
+          showMsg: true,
+          appList: result,
+          staticAppList: result
+        });
+
+        if(that.state.showMsg === true) {
+          setTimeout(that.handleAlertBehaviour, 2000);
+        }
+        
+      }.bind(this),
+      error: function(xhr, status, error) {
+        this.setState({
+          msgBody: "App has not been installed",
+          msgType: "warning",
+          showMsg: true
+        });
+
+        if(that.state.showMsg === true) {
+          setTimeout(that.handleAlertBehaviour, 2000);
+        }
+      }.bind(this),
+      complete: function(result){
         $(":file").filestyle('clear');
-        this.setState({appList: result, staticAppList: result});
-        }.bind(this)
-      })
-    };
+        this.setState({
+          uploadStatus: 0,
+          showProgress: false,
+        });
+      }.bind(this)
+    });
+  }
+
+  handleProgress(e){
+    
+    if(e.lengthComputable){
+      let max = e.total;
+      let current = e.loaded;
+      
+      let Percentage = (current * 100)/max;   
```

Comment:
@paulupendo can we round up the percentage to 1 or 2 decimal places, currently, i'm getting up to 12 decimal places when I upload files. I.e `percentage` = 67.6718736397937399%, I think 67.7% looks better. What do you think?

Reply:
or just 67%

Reply:
@andela-jomadoye @wanjikum I am on the case. 😸